source/puppet/5.5/upgrade_major_server.markdown

```diff
@@ -44,14 +44,24 @@ If you have multiple Puppet masters, upgrade or replace the certificate authorit
 
 Repeat the following steps for each Puppet Server until you're running a pure Puppet 5/Puppet Server 5 infrastructure.
 
-### Install the latest Puppet Server
+### Install the latest Puppet Server 2.x
 
 Our software releases are grouped into the Puppet Platform.
 
 To upgrade Puppet Server, you'll need to add the Puppet Platform repository to each node's package manager. Follow the [Puppet Server installation instructions]({{puppetserver}}/install_from_packages.html) to [enable the Puppet Platform repository](./puppet_platform.html) and install the `puppetserver` package.
 
 Even after you've installed the package, **don't start the `puppetserver` service yet**! You should do a few other things first.
 
+## Upgrade PuppetDB
+
+In the pre-upgrade steps, you upgraded to PuppetDB 2.3.x, including the [`puppetdb-terminus`][puppetdb-terminus] package on your Puppet Servers. This means the upgraded Puppet Server can already communicate with your existing PuppetDB server.
+
+Now that you've upgraded Puppet Server, [upgrade PuppetDB][] to the latest version 3.x. Note that this also retires older API versions, which can break older integrations. See the [PuppetDB 3.0 release notes][] for details.
+
+Use the [`puppetlabs/puppetdb`][puppetdb_module] module to manage your PuppetDB version. Also, note that the terminus package's name is now [`puppetdb-termini`][puppetdb-termini] instead of `puppetdb-terminus`.
+
+After successfully upgrading PuppetDB to version 3, upgrade it to the latest v4 edition of PuppetDB and `puppetdb-termini`.
```

Comment:
Does this need to happen after Puppet Server has been started? (I assume you'd be using `puppet apply` if not?). If so we might want to move the section down a bit? Or split it between what is required before starting Puppet Server and what is done after starting it.

Reply:
@gguillotte Are you happy if I move this below Start Puppet Server? Or do you think it should be split out?

Reply:
I think the general rule has been to upgrade PuppetDB before Puppet Server and the terminus, so that it's ready to handle data/requests from both older and newer server/terminus versions.

I don't think that Puppet Server has to be stopped as far as PuppetDB itself is concerned, you just want to avoid a "too new" Puppet Server sending requests to a PuppetDB that can't handle them.

Reply:
Is it a problem for PuppetDB to be newer than Server? I thought avoiding that "too new" Server problem was part of the point of upgrading PDB + termini together from 2.3.latest to 3.latest, then 4.latest, before starting Server 2.latest.